pandas/core/arrays/arrow/array.py

```diff
@@ -897,9 +898,31 @@ def __setitem__(self, key: int | slice | np.ndarray, value: Any) -> None:
         None
         """
         key = check_array_indexer(self, key)
-        indices = self._indexing_key_to_indices(key)
         value = self._maybe_convert_setitem_value(value)
 
+        # fast path (GH50248)
+        if com.is_null_slice(key):
+            if is_scalar(value) and not pa_version_under6p0:
```

Comment:
Do we need to directly check `not pa_version_under6p0`? If a user has data with a pyarrow data type I think it's implicitly assumed that pyarrow is installed?

Reply:
We recently bumped the min pyarrow version to 6.0 so I think this should be safe to remove

Reply:
done